controllers/controlplane/kopscontrolplane_controller.go

```diff
@@ -114,6 +116,9 @@ type KopsControlPlaneReconciliation struct {
 	kcp            *controlplanev1alpha1.KopsControlPlane
 }
 
+type KCPKey struct{}
```

Comment:
I believe it will be easier to understand using a string instead of a struct here for the keys of the context

Reply:
I used string initially, it fails linting because strings can cause collitions. This is the go idiomatic way to do it and the recommended way

Reply:
go docs: https://pkg.go.dev/context#WithValue
The provided key must be comparable and should not be of type string or any other built-in type to avoid collisions between packages using context. Users of WithValue should define their own types for keys.

Reply:
fair enough!